### ISSUE TITLE
Glowroot does not work with WildFly 15 on Open JDK 13

### ISSUE THREAD

**Issue by carl-payne** [glowroot/glowroot#660]
Recently upgraded our application from WildFly 15 on Open JDK 11 to WildFly 15 on Open JDK 13 and when running Glowroot version: 0.13.3 locally on server hosting our application encounter errors similar to:

2019-11-04 11:21:23.360 ERROR o.g.a.ManagementFactoryHackClassFileTransformer - Unsupported class file major version 57
java.lang.IllegalArgumentException: Unsupported class file major version 57

Full error in attached log file
glowroot.2019-11-04.log

Please review and advise

Regards
Carl

[glowroot.2019-11-04.zip](https://github.com/glowroot/glowroot/files/3803877/glowroot.2019-11-04.zip)

**Comment by atulpendse** [comment]
Facing similar issue with tomcat 8 + jdk 13

```
2020-01-03T04:53:32.809467-08:00 vm_test tomcat[24171]: 2020-01-03 04:53:32.808 ERROR o.g.a.ManagementFactoryHackClassFileTransformer - Unsupported class file major version 57
java.lang.IllegalArgumentException: Unsupported class file major version 57
    at org.glowroot.agent.shaded.org.objectweb.asm.ClassReader.<init>(ClassReader.java:184)
    at org.glowroot.agent.shaded.org.objectweb.asm.ClassReader.<init>(ClassReader.java:166)
    at org.glowroot.agent.shaded.org.objectweb.asm.ClassReader.<init>(ClassReader.java:152)
    at org.glowroot.agent.ManagementFactoryHackClassFileTransformer.transform(ManagementFactoryHackClassFileTransformer.java:48)
    at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:246)
    at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
    at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:563)
    at org.glowroot.agent.MainEntryPoint.premain(MainEntryPoint.java:151)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Nati
```

**Comment by ilyusha** [comment]
We are having the same issue with jetty 9.4 with JDK 13:

```
2020-01-28 22:41:03.147 ERROR o.g.a.ManagementFactoryHackClassFileTransformer - Unsupported class file major version 57
java.lang.IllegalArgumentException: Unsupported class file major version 57
	at org.glowroot.agent.shaded.org.objectweb.asm.ClassReader.<init>(ClassReader.java:184)
	at org.glowroot.agent.shaded.org.objectweb.asm.ClassReader.<init>(ClassReader.java:166)
	at org.glowroot.agent.shaded.org.objectweb.asm.ClassReader.<init>(ClassReader.java:152)
	at org.glowroot.agent.ManagementFactoryHackClassFileTransformer.transform(ManagementFactoryHackClassFileTransformer.java:48)
	at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:246)
	at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:563)
	at org.glowroot.agent.MainEntryPoint.premain(MainEntryPoint.java:151)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.glowroot.agent.AgentPremain.premain(AgentPremain.java:59)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:513)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:525)
```

Does glowroot support JDK 13?

**Comment by meabhi007** [comment]
I am also facing the issue with Javaagent using JDK13.
Is there a way, we can build the javaagent for JDK13?

**Comment by trask** [comment]
Probably the asm dependency needs to be updated

**Comment by meabhi007** [comment]
Thanks for the pointer @trask , Is there any plan to update that dependency and test with JDK13?

**Comment by trask** [comment]
Hi all, this should be fixed now in 0.13.6
